video_manipulation_functions/video_manipulation.py

```diff
@@ -0,0 +1,101 @@
+import cv2
+import shutil
+import os
+from BionicEye.video_manipulation_functions.given_functions import generate_metadata, is_frame_tagged
+from BionicEye.app import Video, Metadata, Frame
+from BionicEye.singelton_classes.db_manager import DBManage
+from BionicEye.singelton_classes.os_manager import OSManage
+
+DB_MANAGE = DBManage()
+OS_MANAGE = OSManage()
+
+
+def count_frames(video_path):
+    """
+    Counts frames in a video
+    :param video_path: path to a video file
+    :return: the amount of frames in the video
+    """
+    frame_number = 0
+    cap = cv2.VideoCapture(video_path)
+    read_correctly, frame = cap.read()
+
+    while read_correctly:
+        frame_number += 1
+        read_correctly, frame = cap.read()
+
+    return frame_number
+
+
+def save_video(video_path):
+    """
+    Saves given video information to the db and the os if not exists already
+    :param video_path: path to the video file we want to save
+    """
+    video_os_path = os.path.relpath(video_path)
+    video_file_name = os.path.basename(video_path)
+    observation_post_name = video_file_name.split('_')[0]
+    frames_amount = count_frames(video_path)
+    if not DB_MANAGE.query(Video.id).filter_by(video_path=video_os_path).first():
+        video = Video(observation_post_name=observation_post_name,
+                      video_path=video_os_path, frames_amount=frames_amount)
+        DB_MANAGE.save(video)
+    OS_MANAGE.upload_file(video_path)
+
+
+def save_metadata(frame):
+    """
+    Save frames metadata to db if not exists
+    :param frame: frame from some video
+    :return: db id of the metadata of the given frame
+    """
+    fov, azimuth, elevation = generate_metadata(frame)
+    frame_tag = is_frame_tagged(frame)
+    metadata_id = DB_MANAGE.query(Metadata.id) \
+        .filter_by(frame_tag=frame_tag, camera_fov=fov, azimuth=azimuth, elevation=elevation) \
```

Comment:
Again `filter_by` selects rows from the entire table instead of stopping on the first result as in `get`